File: frontend/src/api/project.ts
import request from '@/utils/request'

// ============== 枚举常量从 constants 统一导入 ==============
export {
  PROJECT_TYPES,
  PROJECT_PHASES,
  PROJECT_STATUSES
} from './constants'

/**
 * 项目管理 API
 *
 * 后端接口：
 *   GET    /api/projects                        分页查询
 *   GET    /api/projects/{id}                   项目详情
 *   POST   /api/projects                        创建项目（ADMIN/MANAGER）
 *   PUT    /api/projects/{id}                   更新项目
 *   DELETE /api/projects/{id}                   删除项目
 *   GET    /api/projects/{id}/milestones        里程碑列表
 *   POST   /api/projects/{id}/milestones        创建里程碑
 *   DELETE /api/projects/{id}/milestones/{mid}  删除里程碑
 */

export interface ProjectVO {
  id: number
  code: string
  name: string
  type: 'HARDWARE' | 'FIRMWARE' | 'SOFTWARE' | 'MIXED'
  phase?: 'EVT' | 'DVT' | 'PVT' | 'MP'
  managerUserid: string
  managerName?: string
  startDate?: string
  endDate?: string
  actualStartDate?: string
  actualEndDate?: string
  status: 'PLANNING' | 'IN_PROGRESS' | 'ON_HOLD' | 'COMPLETED' | 'CANCELLED'
  progress: number
  description?: string
  tags?: string
  createdBy: string
  createdByName?: string
  createdAt: string
  updatedAt: string
}

export interface MilestoneVO {
  id: number
  projectId: number
  name: string
  phase: 'EVT' | 'DVT' | 'PVT' | 'MP'
  plannedStart: string
  plannedEnd: string
  actualStart?: string
  actualEnd?: string
  progress: number
  dependencies?: string
  ownerUserid?: string
  ownerName?: string
  status: 'NOT_STARTED' | 'IN_PROGRESS' | 'COMPLETED' | 'DELAYED'
  description?: string
  createdAt: string
  updatedAt: string
}

export interface ProjectQuery {
  keyword?: string
  type?: string
  phase?: string
  status?: string
  managerUserid?: string
  pageNum?: number
  pageSize?: number
}

export function listProjects(params: ProjectQuery) {
  return request({ url: '/projects', method: 'get', params })
}

export function getProject(id: number) {
  return request({ url: `/projects/${id}`, method: 'get' })
}

export function createProject(data: Partial<ProjectVO>) {
  return request({ url: '/projects', method: 'post', data })
}

export function updateProject(id: number, data: Partial<ProjectVO>) {
  return request({ url: `/projects/${id}`, method: 'put', data })
}

export function deleteProject(id: number) {
  return request({ url: `/projects/${id}`, method: 'delete' })
}

export function listMilestones(projectId: number) {
  return request({ url: `/projects/${projectId}/milestones`, method: 'get' })
}

export function createMilestone(projectId: number, data: Partial<MilestoneVO>) {
  return request({ url: `/projects/${projectId}/milestones`, method: 'post', data })
}

export function deleteMilestone(projectId: number, milestoneId: number) {
  return request({ url: `/projects/${projectId}/milestones/${milestoneId}`, method: 'delete' })
}

File: backend/src/main/java/com/RD/entity/Project.java
package com.RD.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 项目实体
 *
 * <p>对应表 {@code project}（V1 SQL 3.1 节）</p>
 *
 * <p>字段命名：snake_case → camelCase（{@code manager_userid} → {@code managerUserid}），
 * 跟 spec 文档中的字段名保持一致。</p>
 *
 * <p>枚举字段（type / phase / status）用 {@link String} 存储，
 * 由业务层校验取值范围（避免 MyBatis-Plus 的 {@code IEnum} 跟 ENUM 类型错位）。</p>
 *
 * @author Mavis
 */
@Data
@TableName("project")
public class Project {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 项目编号（唯一） */
    private String code;

    /** 项目名称 */
    private String name;

    /** 项目类型：HARDWARE / FIRMWARE / SOFTWARE / MIXED */
    private String type;

    /** 硬件阶段（仅 HARDWARE/MIXED 使用）：EVT / DVT / PVT / MP */
    private String phase;

    /** 项目负责人 UserID（企微） */
    private String managerUserid;

    /** 计划开始日期 */
    private LocalDate startDate;

    /** 计划结束日期 */
    private LocalDate endDate;

    /** 实际开始日期 */
    private LocalDate actualStartDate;

    /** 实际结束日期 */
    private LocalDate actualEndDate;

    /** 状态：PLANNING / IN_PROGRESS / ON_HOLD / COMPLETED / CANCELLED */
    private String status;

    /** 完成百分比 0-100 */
    private BigDecimal progress;

    /** 描述 */
    private String description;

    /** 标签（JSON 字符串，由业务层序列化/反序列化） */
    private String tags;

    /** 创建人 UserID */
    private String createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /** 逻辑删除标记 */
    @TableLogic
    @TableField("deleted")
    private Integer deleted;
}
